test/gtest/uct/ib/test_ib_xfer.cc

```diff
@@ -96,3 +97,51 @@ UCS_TEST_P(uct_p2p_mix_test_indirect_atomic, mix1000_indirect_atomic,
 
 UCT_INSTANTIATE_IB_TEST_CASE(uct_p2p_mix_test_indirect_atomic)
 
+
+class uct_p2p_mix_test_dm : public uct_p2p_mix_test {
+public:
+    virtual void run(unsigned count) {
+
+        check_run_conditions();
+
+        size_t size = m_send_size;
+        uct_ib_device_mem_h dev_mem;
+        ucs_status_t status;
+        uct_mem_h dm_memh;
+        void *dm_ptr;
+
+        status = uct_ib_md_alloc_device_mem(receiver().md(), &size, &dm_ptr,
```

Comment:
Since it is not public interface, why we gtest this ? Besides it is also has portability issues.

Reply:
we have gtest for many internal functions

Reply:
Generally speaking, original intend of what we discussed was to expose dev header files (pd) and this is it. My expectation was that the allocation, release, and any management code will reside "elsewhere" since it is not directly used in the project. This is something that technically belongs to some other layer. This change goes well beyond of configure and makefile change. Let me think about this. 

For portability reason I suggest to disable this by default and have special option to enable it on the platforms where you care.